source/dump.f90

```diff
@@ -78,37 +67,121 @@ module dump
 ! ================================================================================================================================ !
 subroutine dump_expand_arrays(nele_new, nblz_new)
 
+  use parpro
+  use mod_alloc
   use numerical_constants, only : zero
 
-  implicit none
-
   integer, intent(in) :: nele_new
   integer, intent(in) :: nblz_new
 
-  call alloc(ldump,                 nele_new, .false.,    "ldump",      -1)
-  call alloc(ndumpt,                nele_new, 0,          "ndumpt",     -1)
-  call alloc(dumpfirst,             nele_new, 0,          "dumpfirst",  -1)
-  call alloc(dumplast,              nele_new, 0,          "dumplast",   -1)
-  call alloc(dumpunit,              nele_new, 0,          "dumpunit",   -1)
-  call alloc(dumpfmt,               nele_new, 0,          "dumpfmt",    -1)
-  call alloc(dump_fname, mFileName, nele_new, " ",        "dump_fname", -1)
-
-  call alloc(dumptas,               nblz_new, 6, 6, zero, "dumptas",    -1,1,1)
-  call alloc(dumptasinv,            nblz_new, 6, 6, zero, "dumptasinv", -1,1,1)
-  call alloc(dumpclo,               nblz_new, 6,    zero, "dumpclo",    -1,1)
+  call alloc(ldump,                 nele_new, .false., "ldump",       -1)
+  call alloc(ndumpt,                nele_new, 0,       "ndumpt",      -1)
+  call alloc(dumpfirst,             nele_new, 0,       "dumpfirst",   -1)
+  call alloc(dumplast,              nele_new, 0,       "dumplast",    -1)
+  call alloc(dumpunit,              nele_new, 0,       "dumpunit",    -1)
+  call alloc(dumpfmt,               nele_new, 0,       "dumpfmt",     -1)
+  call alloc(dump_fname, mFileName, nele_new, " ",     "dump_fname",  -1)
+  call alloc(dump_nMatMap,          nele_new, 0,       "dump_nMatMap",-1)
 
 #ifdef CR
-  call alloc(dumpfilepos,           nele_new,-1,          "dumpfilepos",   -1)
-  call alloc(dumpfilepos_cr,        nele_new,-1,          "dumpfilepos_cr",-1)
+  call alloc(dumpfilepos,           nele_new,-1,       "dumpfilepos",   -1)
+  call alloc(dumpfilepos_cr,        nele_new,-1,       "dumpfilepos_cr",-1)
 #endif
 
 #ifdef HDF5
-  call alloc(dump_hdf5DataSet,      nele_new,0,           "dump_hdf5DataSet",-1)
-  call alloc(dump_hdf5Format,       9,       0,           "dump_hdf5Format")
+  call alloc(dump_hdf5DataSet,      nele_new,0,        "dump_hdf5DataSet",-1)
+  call alloc(dump_hdf5Format,       9,       0,        "dump_hdf5Format")
 #endif
 
 end subroutine dump_expand_arrays
 
+! ================================================================================================ !
+!  V.K. Berglyd Olsen
+!  Created: 2019-08-05
+!  Updated: 2019-08-05
+!  Set the normalisation matrix, compute its inverse, and save it in a compact array of structs.
+! ================================================================================================ !
+subroutine dump_setTasMatrix(elemID, tasData, cloData)
+
+  use crcoall
+  use mod_common, only : bez
+  use mod_settings
+
+  integer,          intent(in) :: elemID
```

Comment:
Point out somewhere that this is the *single* element ID, not structure.

Reply:
Done